include/gsl/gsl-lite.hpp

```diff
@@ -1484,21 +1544,32 @@ class not_null
     }
 #endif // gsl_HAVE( RVALUE_REFERENCE )
 
-    gsl_api gsl_constexpr14 get_result_t get() const
-    {
-        return checked_ptr();
-    }
+#if gsl_HAVE( FUNCTION_REF_QUALIFIER )
+# if gsl_CONFIG( NOT_NULL_TRANSPARENT_GET )
+    gsl_api gsl_constexpr14 element_type &  get() const &  { return checked_ptr().get(); }
+    gsl_api gsl_constexpr14 element_type && get() &&       { return std::move(checked_ptr()).get(); }
+# else
+    gsl_api gsl_constexpr14 get_result_t get() const { return checked_ptr(); }
+# endif
 
-#if gsl_HAVE( FUNCTION_REF_QUALIFIER ) && gsl_CONFIG( NOT_NULL_GET_BY_CONST_REF )
-    gsl_api gsl_constexpr   operator T const &     () const & { return checked_ptr(); }
-    gsl_api gsl_constexpr14 operator T &&          () &&      { return std::move(checked_ptr()); }
+    gsl_api gsl_constexpr14 operator T() const & { return checked_ptr(); }
+    gsl_api gsl_constexpr14 operator T() &&      { return std::move(checked_ptr()); }
+
+    gsl_api gsl_constexpr14 T const & operator->() const & { return checked_ptr(); }
+    gsl_api gsl_constexpr14 T &&      operator->() &&      { return std::move(checked_ptr()); }
+
+    gsl_api gsl_constexpr14 element_type &  operator*() const & { return *checked_ptr(); }
+    gsl_api gsl_constexpr14 element_type && operator*() &&      { return *std::move(checked_ptr()); }
```

Comment:
1. Why do we need separate const& and && versions of these operators? Is there any use case when this matters?
2. You changed the return type of operator-> in the "default" (non-constref-get, non-transparent-get) build. Was that your intention?

Reply:
Cf. https://github.com/martinmoene/gsl-lite/commit/a84560ef71816e4b44d5d8a18b34fbf071e23ee6#commitcomment-35452682